toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,368 @@
+	hostname := strings.Trim(n.HostName, " ")
+	re, _ := regexp.Compile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
+
+	if !re.MatchString(hostname) {
+		return fmt.Errorf("Invalid input for --hostname (%s)", hostname)
+	}
+
+	return
+}
+
+func (n *Network) validateIPAddress(ip string) (err error) {
+	bootProto := strings.TrimSpace(n.BootProto)
+	if ip == "" && (bootProto == "" || bootProto == "dhcp") {
+		return
+	} else if net.ParseIP(ip) == nil {
+		return fmt.Errorf("Invalid ip address (%s)", ip)
+	}
+
+	return
+}
+
+// IPAddrIsValid returns an error if ip, gateway, netmask or nameserver inputs are invalid ip addresses
+func (n *Network) IPAddrIsValid() (err error) {
+	ip := strings.Trim(n.Ip, " ")
+	if err = n.validateIPAddress(ip); err != nil {
+		return fmt.Errorf("Invalid input for --ip: %s", err)
+	}
+
+	netmask := strings.Trim(n.NetMask, " ")
+	if err = n.validateIPAddress(netmask); err != nil {
+		return fmt.Errorf("Invalid input for --netmask: %s", err)
+	}
+
+	gateway := strings.Trim(n.GateWay, " ")
+	if err = n.validateIPAddress(gateway); err != nil {
+		return fmt.Errorf("Invalid input for --gateway: %s", err)
+	}
+
+	for _, nameserver := range n.NameServer {
+		if err = n.validateIPAddress(strings.Trim(nameserver, " ")); err != nil {
+			return fmt.Errorf("Invalid input for --nameserver: %s", err)
+		}
+	}
+
+	return
+}
+
+// DeviceIsValid returns an error if the Device name is empty
+func (n *Network) DeviceIsValid() (err error) {
+	if strings.TrimSpace(n.Device) == "" {
+		return fmt.Errorf("Invalid input for --device (), device cannot be empty")
+	}
+	return
+}
+
+func findBootIfValue() (deviceAddr string) {
+	const (
+		kernelArgFile = "/proc/cmdline"
+		startIndex    = 10
+	)
+
+	content, err := os.ReadFile(kernelArgFile)
+	if err != nil {
+		logger.Log.Errorf("failed to read from /proc/cmdline")
+		return
+	}
+
+	// Find the location of BOOTIF
+	kernelArgs := strings.Split(string(content), " ")
+	for _, kernelArg := range kernelArgs {
+		if strings.Contains(kernelArg, "BOOTIF") {
+			rawDeviceAddr := []byte(kernelArg[startIndex:len(kernelArg)])
+
+			// Change the cmdline MAC address into valid format
+			for i, _ := range rawDeviceAddr {
+				if rawDeviceAddr[i] == '-' {
+					rawDeviceAddr[i] = ':'
+				}
+			}
+			deviceAddr = string(rawDeviceAddr)
+			return
+		}
+	}
+
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device = findBootIfValue()
+	}
+
+	for _, iface := range ifaces {
+		if networkData.Device == iface.Name {
+			deviceName = networkData.Device
+			return
+		} else {
+			ifaceAddr := strings.TrimSpace(iface.HardwareAddr.String())
+
+			if strings.Contains(networkData.Device, ":") {
+				networkData.Device = strings.TrimSpace(networkData.Device)
```

Comment:
I'd suggest trimming all values at the very beginning when they are being read from the JSON (unless we need the whitespaces, of course), so we're sure they are always trimmed and we can skip these lines everywhere else in the code.